Assets/Src/Scripts/SeedAvatar/Input/WebCamSource.cs

```diff
@@ -0,0 +1,179 @@
+    public Dropdown FpsDropdown;
+    public Dropdown FlipDropdown;
+    public string DeviceName;
+    public int Fps = 30;
+    public bool Flip = false;
+    [SerializeField] private RenderTexture _buffer;
+    private WebCamTexture _webCam;
+    private int _width = 1280;
+    private int _height = 720;
+    private static readonly Dictionary<int, string> _resolutionDict =
+    new Dictionary<int, string> {
+      { (int)ResolutionOptions.LOW , "1280x720" },
+      { (int)ResolutionOptions.MIDDLE, "1600x900" },
+      { (int)ResolutionOptions.HIGH , "1920x1080" },
+    };
+    private static readonly Dictionary<int, string> _fpsDict =
+    new Dictionary<int, string> {
+      { (int)FpsOptions.LOW , "30Hz" },
+      { (int)FpsOptions.HIGH , "60Hz" },
+    };
+    private static readonly Dictionary<int, string> _flipDict =
+    new Dictionary<int, string> {
+      { (int)FlipOptions.NORMAL , "Normal" },
+      { (int)FlipOptions.MIRRORING , "Mirroring" },
+    };
+    private Dictionary<int, string> _devicesDict = new Dictionary<int, string> { };
+
+    // Start button event.
+    public void Play() {
+      if (_webCam == null) {
+        _webCam = new WebCamTexture(DeviceName, _width, _height, Fps);
+        _webCam.Play();
+      }
+    }
+
+    // Stop button event.
+    public void Stop() {
+      if (_webCam != null) {
+        _webCam.Stop();
+        _webCam = null;
+      }
+    }
+
+    // DropdownChange event.
+    public void OnCameraDropdownChange(int value) {
+      if (_devicesDict[value] != null) {
+        DeviceName = _devicesDict[value];
+      }
+    }
+
+    // DropdownChange event.
+    public void OnResolutionDropdownChange(int value) {
+      switch (value) {
+        case (int)ResolutionOptions.LOW:
+          _width = 1280;
+          _height = 720;
+          break;
+        case (int)ResolutionOptions.MIDDLE:
+          _width = 1600;
+          _height = 900;
+          break;
+        case (int)ResolutionOptions.HIGH:
+          _width = 1920;
+          _height = 1080;
+          break;
+        default:
+          break;
+      }
+    }
+
+    // DropdownChange event.
+    public void OnFpsDropdownChange(int value) {
+      switch (value) {
+        case (int)FpsOptions.LOW:
+          Fps = 60;
+          break;
+        case (int)FpsOptions.HIGH:
+          Fps = 30;
+          break;
+        default:
+          break;
+      }
+    }
+
+    // DropdownChange event.
+    public void OnFlipDropdownChange(int value) {
+      switch (value) {
+        case (int)FlipOptions.NORMAL:
+          Flip = false;
+          break;
+        case (int)FlipOptions.MIRRORING:
+          Flip = true;
+          break;
+        default:
+          break;
+      }
+    }
+
+    private void InitDropdown(Dropdown target, Dictionary<int, string> optDict, UnityAction<int> onChangedCallback) {
+      target.ClearOptions();
+      List<string> optStrings = optDict.Count == 0 ? new List<string>() : new List<string>(optDict.Values);
```

Comment:
`optDict.Values` is retrieving all values from the dictionary. But you are using `Dictionary<int, string>`, instead of `SortedDictionary<int, string>`. The returned collection of `Dictionary.Values` will not be ensured to be ordered by the keys - note that the current code rely on the same order between the 0-n enum values and the option strings to make the dropdown list work correctly.

Since you need to keep the same orders between the integer keys (from 0 to n) and the strings, my last review comments suggested using `SortedDictionary<int, string>` to solve this. Please check the sample code in my last review comments carefully. Thanks,